barretenberg/bbup/bbup.ts

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env node
+import { Command } from "commander";
+const program = new Command();
+import { installBB } from "./shell.js";
+import ora from "ora";
+import logSymbols from "log-symbols";
+import { getBbVersionForNoir } from "./versions.js";
+import { execSync } from "child_process";
+
+const spinner = ora({ color: "blue", discardStdin: false });
+
+const bbup = program
+  .command("install", { isDefault: true })
+  .description("Installs Barretenberg.")
+  .option(
+    "-f, --frontend",
+    "Match the version of a specific frontend language",
+    "noir"
+  );
+
+const options = bbup.opts();
+
+if (options.frontend === "noir") {
+  bbup
+    .requiredOption(
+      "-v, --version <version>",
```

Comment:
```suggestion
      "-nv, --noir-version <version>",
```
Should help mitigate confusions with Barretenberg's own versioning.

Reply:
Yes this makes me think I'll KISS and drop the `-f` flag altogether

Reply:
Ah yeah, that or we can name this option e.g. `-fv, --frontend-version <version>` instead

But that feels unnecessarily confusing until we see more substantial interest in integrating other frontends